frontend/src/scenes/data-warehouse/editor/sidebar/QueryInfo.tsx

```diff
@@ -61,13 +62,26 @@ export function QueryInfo({ codeEditorKey }: QueryInfoProps): JSX.Element {
     const { editingView } = useValues(multitabEditorLogic)
     const { runDataWarehouseSavedQuery, saveAsView } = useActions(multitabEditorLogic)
 
-    const { dataWarehouseSavedQueryMapById, updatingDataWarehouseSavedQuery, initialDataWarehouseSavedQueryLoading } =
-        useValues(dataWarehouseViewsLogic)
-    const { updateDataWarehouseSavedQuery } = useActions(dataWarehouseViewsLogic)
+    const {
+        dataWarehouseSavedQueryMapById,
+        updatingDataWarehouseSavedQuery,
+        initialDataWarehouseSavedQueryLoading,
+        dataModelingJobs,
+        hasMoreJobsToLoad,
+    } = useValues(dataWarehouseViewsLogic)
+    const { updateDataWarehouseSavedQuery, loadDataModelingJobs, loadOlderDataModelingJobs, resetDataModelingJobs } =
+        useActions(dataWarehouseViewsLogic)
 
     // note: editingView is stale, but dataWarehouseSavedQueryMapById gets updated
     const savedQuery = editingView ? dataWarehouseSavedQueryMapById[editingView.id] : null
 
+    useEffect(() => {
+        if (editingView) {
+            resetDataModelingJobs()
+            loadDataModelingJobs(editingView.id)
+        }
+    }, [editingView?.id, loadDataModelingJobs, resetDataModelingJobs])
```

Comment:
We try to avoid using `useEffect` where possible - can this not be moved into the logic? Have a listener on `editingView` or something maybe?

Reply:
I'll see what I can do here. 😄